b9lab-content/3-main-concepts/08-modules.md

```diff
@@ -1,5 +1,243 @@
 # Modules
 
+## Overview
+
+Each Cosmos chain is a purpose-built blockchain and Cosmos SDK modules define the unique properties of each chain. Modules can be considered state machines within the larger state machine. They contain the storage layout, also known as the state, and the state transition functions which are the message methods.
+
+In summary, modules define most of the logic of Cosmos SDK applications.
+
+![transaction message flow to modules](./images/module_overview.png)
+
+When a transaction is relayed from the underlying Tendermint consensus engine, `BaseApp` decomposes the `Messages` contained within the transaction. `BaseApp` routes messages to the appropriate module for processing. Interpretation and execution occurs when the appropriate module message handler receives the message.
+
+Developers compose modules together using the Cosmos SDK to build custom application-specific blockchains.
+
+## Module Scope
+
+Modules include **core** functionality that provides the basic functionality every blockchain node needs:
+
+* a boilerplate implementation of the Application Blockchain Interface (ABCI) that communicates with the underlying Tendermint consensus engine
+* a general-purpose data store that persists the module state called `multistore`
+* a server and interfaces to facilitate interactions with the node.
```

Comment:
```suggestion
* a server and interfaces to facilitate interactions with other modules.
```
Is this supposed to be with other modules? Or what is meant by interactions with the node